------------------------------------------------------------------------
r1439761 | billgraham | 2013-01-29 02:30:29 -0500 (Tue, 29 Jan 2013) | 1 line
Changed paths:
   M /pig/branches/branch-0.11/CHANGES.txt
   M /pig/branches/branch-0.11/src/docs/src/documentation/content/xdocs/perf.xml

PIG-3139: Document reducer estimation (billgraham)

Index: pig/branches/branch-0.11/CHANGES.txt
===================================================================
--- pig/branches/branch-0.11/CHANGES.txt	(revision 1439760)
+++ pig/branches/branch-0.11/CHANGES.txt	(revision 1439761)
@@ -30,6 +30,8 @@
 
 IMPROVEMENTS
 
+PIG-3139: Document reducer estimation (billgraham)
+
 PIG-2341: Need better documentation on Pig/HBase integration (jthakrar and billgraham via billgraham)
 
 PIG-3044: Trigger POPartialAgg compaction under GC pressure (dvryaboy)
Index: pig/branches/branch-0.11/src/docs/src/documentation/content/xdocs/perf.xml
===================================================================
--- pig/branches/branch-0.11/src/docs/src/documentation/content/xdocs/perf.xml	(revision 1439760)
+++ pig/branches/branch-0.11/src/docs/src/documentation/content/xdocs/perf.xml	(revision 1439761)
@@ -172,7 +172,34 @@
 
 </section> 
 
+<!-- ================================================================== -->
+<!-- REDUCER ESTIMATION -->
+<section id="reducer-estimation">
+<title>Reducer Estimation</title>
 
+<p>
+By default Pig determines the number of reducers to use for a given job based on the size of the
+input to the map phase. The input data size is divided by the
+pig.exec.reducers.bytes.per.reducer parameter value (default 1GB) to determine the number of
+reducers. The maximum number of reducers for a job is limited by the pig.exec.reducers.max parameter
+(default 999).
+</p>
+
+<p>
+The default reducer estimation algorithm described above can be overridden by setting the
+pig.exec.reducer.estimator parameter to the fully qualified class name of an implementation of
+<a href="http://svn.apache.org/repos/asf/pig/trunk/src/org/apache/pig/backend/hadoop/executionengine/mapReduceLayer/PigReducerEstimator.java">org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.PigReducerEstimator</a>.
+The class must exist on the classpath of the process submitting the Pig job. If the
+pig.exec.reducer.estimator.arg parameter is set, the value will be passed to a constructor
+of the implementing class that takes a single String.
+</p>
+
+<p>
+</p>
+
+</section>
+
+
 <!-- ==================================================================== -->
 <!-- MULTI-QUERY EXECUTION-->
 <section id="multi-query-execution">

------------------------------------------------------------------------
